packages/langium-cli/src/generator/type-collector.ts

```diff
@@ -126,25 +167,100 @@ class TypeTree {
     }
 }
 
-export function collectAst(documents: LangiumDocuments, grammars: langium.Grammar[]): Interface[] {
-    const state = createState();
+class StateCollector {
```

Comment:
Suggestion: This name is quite misleading. This class is not a collector of states, but represents the state of the collector and should therefore be named `(Type)CollectorState`.